File: SurveyAPI/SurveyAPI/Models/QuestionSave.cs
namespace SurveyAPI.Models {
    public class QuestionSave {
        public string Name { get; set; } = string.Empty;
        public DateTime BeginDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool Multiple { get; set; } = false;
        public List<ChoiceSave> Choices { get; set; } = new List<ChoiceSave>();
    }
}

File: front-vue/src/models/question/QuestionSave.ts
import { ChoiceSave, ChoiceSaveDTO } from "./ChoiceSave";

export class QuestionSaveDTO {
  name = "";
  beginDate = "";
  endDate = "";
  multiple = false;
  choices: ChoiceSaveDTO[] = [];
}

export class QuestionSave extends QuestionSaveDTO {
  constructor(data: QuestionSave | QuestionSaveDTO | null) {
    super();
    if (data) {
      Object.assign(this, data);
      this.choices = data.choices.map((c) => new ChoiceSave(c));
    }
  }
}
